File: golang/algorithms/minion_game.go
package algorithms

import (
	"fmt"
	"unicode/utf8"
)

func MinionGame(str string) {
	vowels := []uint8{'A', 'E', 'I', 'O', 'U'}
	kevsc, stusc := 0, 0

	ln := utf8.RuneCount([]byte(str))

	for i := 0; i < ln-1; i++ {
		if ExistsInArrayString(str[i], vowels) {
			kevsc += (ln - i)
		} else {
			stusc += (ln - i)
		}
	}

	if kevsc > stusc {
		fmt.Printf("Kevin %v\n", kevsc)
	} else {
		fmt.Printf("Stuart %v\n", stusc)
	}
}

func ExistsInArrayString(value uint8, slice []uint8) bool {
	for _, item := range slice {
		if item == value {
			return true
		}
	}

	return false
}

File: python/algorithms/minion_game.py
def minion_game(string):
    vowels = ["A", "E", "I", "O", "U"]
    kevsc, stusc = 0, 0
    for i in range(len(string)):
        if s[i] in vowels:
            kevsc += len(s) - i
        else:
            stusc += len(s) - i

    if kevsc > stusc:
        print(f"Kevin {kevsc}")
    elif kevsc < stusc:
        print(f"Stuart {stusc}")
    else:
        print("Draw")


if __name__ == "__main__":
    s = input()
    minion_game(s)
